posthog/models/filters/mixins/funnel_trends.py

```diff
@@ -0,0 +1,24 @@
+from posthog.constants import FUNNEL_FROM_STEP, FUNNEL_TO_STEP
+from posthog.models.filters.mixins.base import BaseParamMixin
+from posthog.models.filters.mixins.utils import cached_property, include_dict
+
+
+class FunnelTrendsMixin(BaseParamMixin):
+    @cached_property
+    def funnel_from_step(self) -> int:
+        from_step = int(self._data.get(FUNNEL_FROM_STEP, 1))
+        return from_step
+
+    @cached_property
+    def funnel_to_step(self) -> int:
+        to_step = int(self._data.get(FUNNEL_TO_STEP, len(self.entities)))
+        return to_step
+
+    @include_dict
+    def funnel_window_days_to_dict(self):
```

Comment:
typo xD

Reply:
Ah, right, that's my keyboard's fault ofc:

![](https://149351115.v2.pressablecdn.com/wp-content/uploads/2021/03/april-fools-social-2048x1072.jpg)